src/components/UserSelector.tsx

```diff
@@ -20,21 +60,23 @@
 
       <div className="dropdown-menu" id="dropdown-menu" role="menu">
         <div className="dropdown-content">
-          <a href="#user-1" className="dropdown-item">
-            Leanne Graham
-          </a>
-          <a href="#user-2" className="dropdown-item is-active">
-            Ervin Howell
-          </a>
-          <a href="#user-3" className="dropdown-item">
-            Clementine Bauch
-          </a>
-          <a href="#user-4" className="dropdown-item">
-            Patricia Lebsack
-          </a>
-          <a href="#user-5" className="dropdown-item">
-            Chelsey Dietrich
-          </a>
+          {users.map(user => (
+            <a
+              href={`#user-${user.id}`}
+              key={user.id}
+              className={classNames('dropdown-item', {
+                'is-active': user.id === userId,
+              })}
+              onClick={() => {
+                setUserId(user.id);
+                setUserName(user.name);
+                setShowList(false);
+                setSelectedpost(null);
```

Comment:
The function `setSelectedpost` should be renamed to `setSelectedPost` to maintain consistency with the naming convention used for state variables and props.